example/sine.zig

```diff
@@ -132,18 +138,92 @@ fn streamStateCallback(stream: *pa.stream, userdata: ?*anyopaque) callconv(.c) v
 
 fn streamWriteCallback(stream: *pa.stream, requested_bytes: usize, userdata: ?*anyopaque) callconv(.c) void {
     const pulse: *Pulse = @ptrCast(@alignCast(userdata));
-    _ = pulse;
+    const sample_spec = stream.get_sample_spec();
     std.log.info("requested bytes: {d}", .{requested_bytes});
     var remaining_bytes = requested_bytes;
     while (remaining_bytes > 0) {
         var ptr_len: usize = remaining_bytes;
-        var opt_ptr: ?[*]i32 = null;
```

Comment:
note, that this line seemed to intermittently be causing the following assert:
```
Assertion 'pa_atomic_load(&(b)->_ref) > 0' failed at /home/marler8997/git/pulseaudio/src/pulsecore/memblock.c:682, function void pa_memblock_unref(pa_memblock *)(). Aborting.
Aborted (core dumped)
```
Changing the pointer type to `?[*]u8` seems to fix it, so I'm guessing this might be triggering some sort of alignment issue and the assert is probably bogus.